### PR TITLE
[WIP] Migrate cumsum from the TH to Aten (CPU)

### PR DESCRIPTION
**Pull request by ifedan**

### CODE REVIEW

File: aten/src/ATen/native/cpu/UnaryOpsKernel.cpp [hunk 1]
```diff
@@ -322,6 +323,57 @@ static void rsqrt_kernel(TensorIterator& iter) {
...
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "cumsum_cpu", [&] {
+    using accscalar_t = at::acc_type<scalar_t, false>;
+    const scalar_t * const self_ptr = self.data_ptr<scalar_t>();
+    scalar_t * const res_ptr = result.data_ptr<scalar_t>();
+    int64_t d_size = self.size(dimension);
+    int64_t self_stride = self.stride(dimension);
+    int64_t res_stride = self.stride(dimension);
+    int64_t total = self.numel() / d_size;
+
+    parallel_for(0, total, internal::GRAIN_SIZE, [&](int64_t s, int64_t e) {
```
Comment (andreaskoepf) [inline aten/src/ATen/native/cpu/UnaryOpsKernel.cpp]: just a question: when do we use parallel_for and when parallel_reduce ?

Comment (ifedan) [reply aten/src/ATen/native/cpu/UnaryOpsKernel.cpp]: parallel_reduced is used for reduction operation in TensorIterator

File: aten/src/ATen/native/UnaryOps.cpp [hunk 2]
```diff
@@ -235,6 +235,28 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
...
+  if (self.dim() < 1) {
+    result.copy_(self);
+  } else {
+    cumsum_stub(device1, result, self, dim);
+  }
+  return result;
+}
+
+Tensor cumsum_cpu(const Tensor& self, int64_t dimension) {
+  Tensor result = at::empty_like(self);
```
Comment (andreaskoepf) [inline aten/src/ATen/native/UnaryOps.cpp]: Should we use the input dtype or better go for an accumulator-type?
e.g. the old behavior was
```
>>> x = torch.randint(0, 255, (100,), dtype=torch.int8)
>>> x.type()
'torch.CharTensor'
>>> x.cumsum(0).type()
'torch.LongTensor'
```

Comment (ifedan) [reply aten/src/ATen/native/UnaryOps.cpp]: @andreaskoepf thanks you for you comments, this PR still in progress. It's draft and has [WIP] (work in progress) tag.
I think you are right, accscalar_t is a good idea.

**Pull request closed without merging**